lib/git/git.js

```diff
@@ -0,0 +1,310 @@
+    data.branch = match ? match[1] : 'HEAD';
+
+    return cb(null, data);
+  });
+};
+
+
+git.getRemote = function (folder, data, cb) {
+  git.parseGitConfig(folder, function (err, config) {
+    if (err) {
+      return cb(err);
+    }
+
+    data.remotes = [];
+
+    _.forEach(config, function (value, key) {
+      var regex = /remote "(.*)"/;
+      var match = regex.exec(key);
+      if (match) {
+        data.remotes.push(match[1]);
+      }
+    });
+
+    data.remote = (data.remotes.indexOf('origin') === -1) ? data.remotes[0] : 'origin';
+
+    cb(null, data);
+  });
+};
+
+git.isCurrentBranchOnRemote = function (folder, data, cb) {
+  jsGitService.getRefHash(folder, data.branch, data.remote, function (err, hash) {
+    if (err) {
+      return cb(err);
+    }
+
+    data.branch_exists_on_remote = !!hash;
+
+    return cb(null, data);
+  });
+};
+
+git.getPrevNext = function (folder, data, cb) {
+  var remote = data.branch_exists_on_remote ? data.remote : null;
+
+  jsGitService.getCommitHistory(folder, 100, data.branch, remote, function (err, commitHistory) {
+    if (err) {
+      return cb(err);
+    }
+
+    var currentCommitIndex = _.findIndex(commitHistory, {hash: data.revision});
+
+    if (currentCommitIndex === -1) {
+      data.ahead = true;
+      data.next_rev = null;
+      data.prev_rev = null;
+    }
+    else {
+      data.ahead = false;
+      data.next_rev = (currentCommitIndex === 0) ? null : commitHistory[currentCommitIndex - 1].hash;
+      data.prev_rev = (currentCommitIndex === (commitHistory.length - 1)) ? null : commitHistory[currentCommitIndex + 1].hash;
+    }
+
+    cb(null, data);
+  });
+};
+
+git.getUpdateTime = function (folder, data, cb) {
+  fs.stat(folder + ".git", function (err, stats) {
+    if (err) {
+      return cb(err);
+    }
+
+    data.update_time = helper.trimNewLine(stats.mtime);
+    return cb(null, data);
+  });
+};
+
+git.getTags = function (folder, data, cb) {
+  exec("cd '" + folder + "';LC_ALL=en_US.UTF-8 git tag", {timeout: TIMEOUT, maxBuffer: MAXBUFFER},
+    function (err, stdout, stderr) {
+      if (err) {
+        return cb(err);
+      }
+
+      if (stdout.length) {
+        data.tags = stdout.split('\n');
+        data.tags.pop();
+        data.tags = data.tags.slice(0, 10);
+      }
+      return cb(null, data);
+    });
+};
+
+git.parse = function (folder, cb) {
+  async.waterfall([
+      git.getUrl.bind(null, folder),
+      git.getCommitInfo.bind(null, folder),
+      git.getStaged.bind(null, folder),
+      git.getBranch.bind(null, folder),
+      git.getRemote.bind(null, folder),
+      git.isCurrentBranchOnRemote.bind(null, folder),
+      git.getPrevNext.bind(null, folder),
+      git.getUpdateTime.bind(null, folder),
+      git.getTags.bind(null, folder)],
+    function (err, data) {
+      if (err) {
+        return cb(err);
+      }
+
+      return cb(null, data);
+    });
+};
+
+git.isUpdated = function (folder, cb) {
+  git.parse(folder, function (err, data) {
```

Comment:
Je doute qu'on ai besoin de recuperer toutes les datas pour verifier que le repo est up-to-date, sa serait cool de reduire au minimum de call possible chaque fonction.